apps/model-diagnostics/model_diagnostics.cpp

```diff
@@ -0,0 +1,54 @@
+/*************************************************
+USAGE:
+./model_diagnostics -m <onnx file location>
+**************************************************/
+#include <opencv2/dnn.hpp>
+#include "../../samples/dnn/common.hpp"
+
+#include <fstream>
+#include <iostream>
+
+
+using namespace cv;
+using namespace dnn;
+
+
+static void diagnosticsErrorCallback(const Exception& exc)
+{
+    CV_UNUSED(exc);
+    fflush(stdout);
+    fflush(stderr);
+}
+
+std::string diagnosticKeys =
+        "{ model m     | | Path to the model .onnx file. }"
+        "{ config c    | | Path to the model configuration file. }"
+        "{ framework f | | [Optional] Name of the model framework. }";
+
+
+
+int main( int argc, const char** argv )
+{
+    CommandLineParser argParser(argc, argv, diagnosticKeys);
+    argParser.about("Use this tool to run the diagnostics of provided ONNX model"
+                    "to obtain the information about its support (supported layers).");
+
+    if (argc == 1)
+    {
+        argParser.printMessage();
+        return 0;
+    }
+
+    String model = findFile(argParser.get<String>("model"));
```

Comment:
I do not think that `findFile` call is good idea. It's diagnostic tool and it should work with path without search.